thunder/tests/test_grad.py

```diff
@@ -1355,7 +1360,7 @@ def filter_differentiable_outputs(outputs):
             grads = [torch.ones_like(torch_result)]
 
     torch_tensors_requiring_grad = tuple(f for f in torch_flats if isinstance(f, torch.Tensor) and f.requires_grad)
-    torch_grad_result = torch.autograd.grad(torch_result, torch_tensors_requiring_grad, grads)
+    torch_grad_result = torch.autograd.grad(torch_result, torch_tensors_requiring_grad, grads, allow_unused=True)
```

Comment:
Why is this change needed?

Reply:
I was getting something like `differentiated tensor is not used in the graph` earlier. I don't exactly remember, but it looks like I cannot reproduce this error anymore. Thanks for pointing it out; I've updated the PR.